### PR TITLE
Replace GEOSPreparedContains with internal winding number algorithm

### PR DESCRIPTION
**Pull request by bjornharrtell**
Replica of https://git.osgeo.org/gitea/postgis/postgis/pulls/24 for CI.

### CODE REVIEW

File: topology/test/perf/TopoGeo_addLinestring.sql [hunk 1]
```diff
@@ -8,3 +8,5 @@ SELECT count(*) FROM (
       generate_series(-15,15,5) y
   ) foo
 ) bar;
+\timing
```
Comment (Komzpa) [inline topology/test/perf/TopoGeo_addLinestring.sql]: can it be explicit `\timing on` or `\timing off`?

**Comment by Komzpa** [comment]
@bjornharrtell can you also update NEWS and post a link to trac ticket?

File: liblwgeom/lwgeom_topo.c [hunk 2]
```diff
@@ -616,46 +615,27 @@ _lwt_CheckEdgeCrossing( LWT_TOPOLOGY* topo,
...
-    GEOSGeom_destroy(nodegg);
-    if (contains == 2)
-    {
-      GEOSPreparedGeom_destroy(prepared_edge);
-      GEOSGeom_destroy(edgegg);
-      _lwt_release_nodes(nodes, num_nodes);
-      lwerror("GEOS exception on PreparedContains: %s", lwgeom_geos_errmsg);
-      return -1;
-    }
+    const POINT2D *pt = getPoint2d_cp(node->geom->point, 0);
```
Comment (Algunenano) [inline liblwgeom/lwgeom_topo.c]: For the sake of consistency, could you move variable declarations to the top of their block? Same for https://github.com/bjornharrtell/postgis/blob/950bf4743c97b9f61b501914d4d36febb1de35cc/liblwgeom/lwgeom_topo.c#L3322 and https://github.com/bjornharrtell/postgis/blob/950bf4743c97b9f61b501914d4d36febb1de35cc/liblwgeom/lwgeom_topo.c#L5046

**Comment by strk** [comment]
Agreed, variables declared at start of block, for ansi compatibility

**Comment by strk** [comment]
> can it be explicit `\timing on` or `\timing off`?

Yes, but I think \timing should just go away, replaced with
timing done by run_test.pl itself (when invoked with --perf
or something similar)

**Comment by Komzpa** [comment]
@bjornharrtell what's left to do announce this finished?

**Comment by bjornharrtell** [comment]
@Komzpa - I view this as an experiment on which I wanted feedback on the method itself before iterating it to a finished state with a proper issue at Trac and NEWS addition. I also think it needs a wider set of performance tests to verify it's really faster, I'm not sure it is for edges with large amounts of vertices. I added bbox contains short circuit too but removed again as I could not measure any improvement, but might be relevant for other topologies.

**Comment by Komzpa** [comment]
oh, then you should request feedback explicitly, or we're going to think you know what you are doing! :)

personally I like how many lines you have removed without breaking any test, that is usually a sign of good change.

**Comment by Komzpa** [comment]
@bjornharrtell I was going to merge this but see that you have commit rights. Can you merge this then? :)

**Comment by bjornharrtell** [comment]
Ok, will do in near future. :)

File: liblwgeom/lwgeom_topo.c (hunk 2 above)
Comment (bjornharrtell) [reply liblwgeom/lwgeom_topo.c]: hmm but that means I have to drop const. :( oh well.

Comment (bjornharrtell) [reply liblwgeom/lwgeom_topo.c]: um.. no it doesn't :) C is wierd..

**Comment by Komzpa** [comment]
Thank you!

**Pull request closed without merging**